### PR TITLE
Monitor elasticsearch metric filtering

### PR DESCRIPTION
**Pull request by bjsignalfx**

### CODE REVIEW

File: internal/monitors/elasticsearch/client/metricnames.go [hunk 1]
```diff
@@ -35,44 +35,44 @@ const (
 	// Flush stats
 	indicesFlushTotal     = "elasticsearch.indices.flush.total"
 	indicesFlushPeriodic  = "elasticsearch.indices.flush.periodic"
-	indicesFlushTotalTime = "elasticsearch.indices.flush.total-time"
+	IndicesFlushTotalTime = "elasticsearch.indices.flush.total-time"
```
Comment (jaysfx) [inline internal/monitors/elasticsearch/client/metricnames.go]: Remove all these where possible and use the generated constants from genmetadata.go. If there are metrics missing from metadata add them.

File: internal/monitors/elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -319,6 +319,139 @@ func (m *Monitor) sendDatapoints(dps []*datapoint.Datapoint) {
...
+	}
+	if c.EnableEnhancedTransportStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeTransport]...)
+	}
+	enhancedStatsForIndexGroups := utils.StringSliceToMap(append(c.EnableEnhancedNodeStatsForIndexGroups, c.EnableEnhancedIndexStatsForIndexGroups...))
+	if enhancedStatsForIndexGroups[client.StoreStatsGroup] {
+		extraMetrics = append(extraMetrics, elasticsearchIndicesStoreThrottleTime)
+	}
+	if enhancedStatsForIndexGroups[client.IndexingStatsGroup] {
+		extraMetrics = append(extraMetrics, elasticsearchIndicesIndexingIndexCurrent)
```
Comment (jaysfx) [inline internal/monitors/elasticsearch/elasticsearch.go]: Do these not relate to the groups somehow? We ideally wouldn't have to hardcode so many metrics here.

File: internal/monitors/elasticsearch/elasticsearch.go [hunk 3]
```diff
@@ -319,6 +319,139 @@ func (m *Monitor) sendDatapoints(dps []*datapoint.Datapoint) {
...
+		extraMetrics = append(extraMetrics, elasticsearchIndicesGetExistsTime)
+		extraMetrics = append(extraMetrics, elasticsearchIndicesGetMissingTotal)
+		extraMetrics = append(extraMetrics, elasticsearchIndicesGetMissingTime)
+	}
+	if enhancedStatsForIndexGroups[client.SearchStatsGroup] {
+		extraMetrics = append(extraMetrics, elasticsearchIndicesSearchQueryCurrent)
+		extraMetrics = append(extraMetrics, elasticsearchIndicesSearchFetchCurrent)
+		extraMetrics = append(extraMetrics, elasticsearchIndicesSearchScrollCurrent)
+		extraMetrics = append(extraMetrics, elasticsearchIndicesSearchSuggestCurrent)
+		extraMetrics = append(extraMetrics, client.IndicesSearchOpenContexts)
```
Comment (jaysfx) [inline internal/monitors/elasticsearch/elasticsearch.go]: Looks like these metrics from client should all be in the metadata.

**Comment by jaysfx** [comment]
Also see seon's docker change that just got merged. It uses .EnabledMetrics to determine which groups should be implicitly turned on as well.

File: internal/monitors/elasticsearch/client/metricnames.go (hunk 1 above)
Comment (jaysfx) [reply internal/monitors/elasticsearch/client/metricnames.go]: Is this possible?

Comment (bjsignalfx) [reply internal/monitors/elasticsearch/client/metricnames.go]: No, because the metadata constants are not exported so can't use them in package client.

**Pull request merged**